staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -675,7 +595,17 @@ See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts`
 		if !errors.IsNotFound(err) {
 			return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%s\nfrom server for:", info.String()), info.Source, err)
 		}
+		// If the object doesn't exist and serverSideAuto is set
+		// server-side apply
+		if o.ServerSideApplyStrategy == cmdutil.ServerSideAuto {
+			return o.serverSideApplyOneObject(info, helper, originalData)
+		}
 
+		if o.FieldManager == fieldManagerUnset {
+			helper = helper.WithFieldManager(FieldManagerClientSideApply)
```

Comment:
Nit: A comment at this point saying that we've eliminated all the values other than ServerSideApplyStrategy ClientSide would be helpful

Personally, I would enforce that in code with a switch statement (and panic / return an error on `default`), but a comment would suffice

Reply:
Looking closely I think it wouldn't ever actually hit `helper = helper.WithFieldManager(FieldManagerClientSideApply)` because this is the case where it is a new object, and we know the strategy is not `auto` which means it would have not have been possible for it to be set to `fieldManagerUnset` with the current implementation.

Reply:
> we've eliminated all the values other than ServerSideApplyStrategy ClientSide would be helpful
It's not that we've eliminated possible values passed in so much as we are at a point in the code where we're sure which _default_ to apply when no value was passed in (i.e. when o.FieldManager == fieldManagerUnset). Before, we could always apply the default at flag parsing time, but now it needs to be applied on a per-object bases (because some objects need the SSA default and other objects need the CSA default).